mne/decoding/tests/test_base.py

```diff
@@ -92,39 +133,35 @@ def inverse_transform(self, X):
     patterns_inv = get_coef(clf, 'patterns_', True)
     assert_true(patterns[0] != patterns_inv[0])
 
-    # Check patterns values
-    clf = make_pipeline(StandardScaler(), LinearModel(LinearRegression()))
-    clf.fit(X, y)
-    patterns = get_coef(clf, 'patterns_', True)
-    mean, std = X.mean(0), X.std(0)
-    X = (X - mean) / std
-    coef = np.linalg.pinv(X.T.dot(X)).dot(X.T.dot(y))
-    patterns_manual = np.cov(X.T).dot(coef)
-    assert_array_almost_equal(patterns, patterns_manual * std + mean)
-
     # Check with search_light and combination of preprocessing ending with sl:
-    n_samples, n_features, n_times = 20, 3, 5
-    y = np.arange(n_samples) % 2
-    X = np.random.rand(n_samples, n_features, n_times)
-    slider = SlidingEstimator(make_pipeline(StandardScaler(), LinearModel()))
-
+    slider = SlidingEstimator(make_pipeline(StandardScaler(), lm))
+    X = np.transpose([X, -X], [1, 2, 0])  # invert X across 2 time samples
     clfs = (make_pipeline(Scaler(None, scalings='mean'), slider), slider)
     for clf in clfs:
         clf.fit(X, y)
         for inverse in (True, False):
             patterns = get_coef(clf, 'patterns_', inverse)
             filters = get_coef(clf, 'filters_', inverse)
             assert_array_equal(filters.shape, patterns.shape,
-                               [n_features, n_times])
+                               X.shape[1:])
+            # the two time samples get inverted patterns
+            assert_equal(patterns[0, 0], -patterns[0, 1])
     for t in [0, 1]:
         assert_array_equal(get_coef(clf.estimators_[t], 'filters_', False),
                            filters[:, t])
 
+    # Check patterns with more than 1 regressor
+    X, Y, A = _make_data(n_samples=2000, n_targets=2)
+    lm.fit(X, Y)
+    assert_array_equal(lm.filters_.shape, lm.patterns_.shape, [2, 5])
+    assert_array_equal(np.round(A * 10), np.round(lm.patterns_.T * 10))
```

Comment:
This should do the same, but clearer:

    assert_allclose(A, lm.patterns_.T, rtol=0.1)